crates/oro-client/src/credentials.rs

```diff
@@ -0,0 +1,49 @@
+use std::{collections::HashMap, fmt::Debug};
+
+use crate::OroClientError;
+
+/**
+ * HTTP basic credentials (i.e. username & password)
+ */
+#[derive(Clone)]
+pub struct BasicAuthCredentials {
+    pub username: String,
+    pub password: String,
+}
+
+impl Debug for BasicAuthCredentials {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        f.write_fmt(format_args!("username: {}, password: ***", self.username))
+    }
+}
+
+/**
+ * Different credential types supported by orogene
+ */
+#[derive(Clone, Debug)]
+pub enum Credentials {
+    Basic(BasicAuthCredentials),
+    Token(String),
+}
+
+impl TryFrom<&HashMap<&String, &String>> for Credentials {
+    type Error = OroClientError;
+
+    fn try_from(value: &HashMap<&String, &String>) -> Result<Self, Self::Error> {
+        // TODO I don't like those &...to_string() constructs -> find a better type for the HashMap!
+        if let Some(&token) = value.get(&"token".to_string()) {
```

Comment:
doesn't `value.get("token")` work as-is?...

Reply:
Not for `HashMap<&String, &String>`. For `HashMap<String, String>`, however... I've changed the type of the map.